port/common/omrcuda.cpp

```diff
@@ -292,84 +292,200 @@ J9CUDA_STATIC_ASSERT(
+	 * they are potentially associated with a version of the runtime which is newer
+	 * than what is available at compile time.
+	 */
+	{ J9CUDA_ERROR_MISSING_CONFIGURATION, 52 },
+	{ J9CUDA_ERROR_LAUNCH_FAILURE, 719 },
+	{ J9CUDA_ERROR_LAUNCH_TIMEOUT, 702 },
+	{ J9CUDA_ERROR_LAUNCH_OUT_OF_RESOURCES, 701 },
+	{ J9CUDA_ERROR_INVALID_DEVICE_FUNCTION, 98 },
+	{ J9CUDA_ERROR_INVALID_DEVICE, 101 },
+	{ J9CUDA_ERROR_INVALID_VALUE, 1 },
+	{ J9CUDA_ERROR_MAP_BUFFER_OBJECT_FAILED, 205 },
+	{ J9CUDA_ERROR_UNMAP_BUFFER_OBJECT_FAILED, 206 },
+	{ J9CUDA_ERROR_CUDART_UNLOADING, 4 },
+	{ J9CUDA_ERROR_UNKNOWN, 999 },
+	{ J9CUDA_ERROR_INVALID_RESOURCE_HANDLE, 400 },
+	{ J9CUDA_ERROR_NOT_READY, 600 },
+	{ J9CUDA_ERROR_SET_ON_ACTIVE_PROCESS, 708 },
+	{ J9CUDA_ERROR_NO_DEVICE, 100 },
+	{ J9CUDA_ERROR_ECCUNCORRECTABLE, 214 },
+	{ J9CUDA_ERROR_SHARED_OBJECT_SYMBOL_NOT_FOUND, 302 },
+	{ J9CUDA_ERROR_SHARED_OBJECT_INIT_FAILED, 303 },
+	{ J9CUDA_ERROR_UNSUPPORTED_LIMIT, 215 },
+	{ J9CUDA_ERROR_INVALID_KERNEL_IMAGE, 200 },
+	{ J9CUDA_ERROR_NO_KERNEL_IMAGE_FOR_DEVICE, 209 },
+	{ J9CUDA_ERROR_PEER_ACCESS_ALREADY_ENABLED, 704 },
+	{ J9CUDA_ERROR_PEER_ACCESS_NOT_ENABLED, 705 },
+	{ J9CUDA_ERROR_DEVICE_ALREADY_IN_USE, 216 },
+	{ J9CUDA_ERROR_PROFILER_DISABLED, 5 },
+	{ J9CUDA_ERROR_ASSERT, 710 },
+	{ J9CUDA_ERROR_TOO_MANY_PEERS, 711 },
+	{ J9CUDA_ERROR_HOST_MEMORY_ALREADY_REGISTERED, 712 },
+	{ J9CUDA_ERROR_HOST_MEMORY_NOT_REGISTERED, 713 },
+	{ J9CUDA_ERROR_OPERATING_SYSTEM, 304 },
+	{ J9CUDA_ERROR_PEER_ACCESS_UNSUPPORTED, 217 },
+	{ J9CUDA_ERROR_NOT_PERMITTED, 800 },
+	{ J9CUDA_ERROR_NOT_SUPPORTED, 801 }
+};
+
+/**
+ * Translate the given runtime API error to an OMR error code.
+ *
+ * @param[in] portLibrary the port library pointer
+ * @param[in] error a CUDA runtime API error code
+ * @return an equivalent OMR error code
+ */
+J9CudaError
+mapRuntime(OMRPortLibrary *portLibrary, cudaError_t error)
+{
+	if (cudaSuccess == error) {
+		/* short-circuit the most common case */
+		return J9CUDA_NO_ERROR;
+#if CUDA_VERSION < 10010
+	} else if (portLibrary->portGlobals->cudaGlobals.runtimeVersion < 10010) {
+		/*
+		 * No mapping is required if the CUDA runtime version is before 10.1.
+		 * If we're compiling with 10.1 or later, that can't happen because
+		 * we refuse to load a runtime library whose version is less than
+		 * the compile-time version.
+		 */
+#endif /* CUDA_VERSION < 10010 */
+	} else {
+		const RuntimeCodePair *pair = newRuntimeCodePairs;
+
+		for (size_t i = 0; i < LENGTH_OF(newRuntimeCodePairs); ++i, ++pair) {
+			if ((int32_t)error == (int32_t)pair->cudaCode) {
+				return pair->omrCode;
+			}
+		}
+	}
+
+	return (J9CudaError)(int32_t)error;
+}
+
+/**
+ * Translate the given OMR error code to a CUDA runtime API error code.
+ *
+ * @param[in] portLibrary the port library pointer
+ * @param[in] error an OMR error code
+ * @return an equivalent CUDA error code
+ */
+cudaError_t
+unmapRuntime(OMRPortLibrary *portLibrary, J9CudaError error)
```

Comment:
I don't think `unmap` is the best choice of word here since it is just another mapping, from `J9CudeError` to `cudaError_t`.

Reply:
This does the opposite of `mapRuntime` so `unmapRuntime` seems reasonable to me.
The former maps from CUDA runtime codes to OMR codes.